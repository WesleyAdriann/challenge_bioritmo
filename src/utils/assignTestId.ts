type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
)  => ({
    'data-testid': `${componentType}_${componentId}`,
  })


export const mockTestID = (
    componentType: componentType,
    componentId: string
  ) => assignTestId(componentType, componentId)
