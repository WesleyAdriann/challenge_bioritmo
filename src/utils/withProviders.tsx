
import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../style'

export const StorybookProviders: React.FC = ({
  children,
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
)
