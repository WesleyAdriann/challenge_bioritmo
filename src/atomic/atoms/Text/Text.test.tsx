import { render, cleanup} from '@testing-library/react'

import { mockTestID, StorybookProviders as wrapper } from '../../../utils'

import { Text, IText } from './index'

describe('Atoms/Button', () => {
  const props: IText = {
    testID: 'Text',
    children: 'Text Test'
  }

  const renderComponent = () => render(
    <Text {...props} />, { wrapper }
  )

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const el_container = mockTestID('text', props.testID!)

  it(`should render component #${el_container}`, async () => {
    const sut = await component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })

})
