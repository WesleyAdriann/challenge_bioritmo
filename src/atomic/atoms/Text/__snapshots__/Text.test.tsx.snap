// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button should render snapshot 1`] = `
.c0 {
  color: #000;
  font-weight: 400;
  font-size: 14px;
  text-align: left;
}

<p
  class="c0"
  data-testid="text_Text"
>
  Text Test
</p>
`;
